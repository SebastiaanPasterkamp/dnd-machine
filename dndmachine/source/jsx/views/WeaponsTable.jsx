import React from 'react';
import Reflux from 'reflux';

import listDataActions from '../actions/listDataActions.jsx';
import DataStore from '../stores/dataStore.jsx';

import LazyComponent from '../components/LazyComponent.jsx';
import Damage from '../components/Damage.jsx';
import Reach from '../components/Reach.jsx';
import Coinage from '../components/Coinage.jsx';

class WeaponsHeader extends LazyComponent
{
    render() {
        return <thead>
            <tr>
                <th>{this.props.name}</th>
                <th>Damage</th>
                <th>Range</th>
                <th>Cost</th>
                <th>Properties</th>
            </tr>
        </thead>
    }
};

class WeaponsRow extends LazyComponent
{
    render() {
        return <tr
                data-name={this.props.name}>
            <td>{this.props.name}</td>
            <td><Damage {...this.props.damage}/></td>
            <td>{this.props.range
                ? <Reach {...this.props.range}/>
                : null
            }</td>
            <td>{this.props.cost
                ? <Coinage {...this.props.cost} extended="1" />
                : null
            }</td>
            <td></td>
        </tr>
    }
};

class WeaponsBody extends LazyComponent
{
    filterItem(item) {
        return (
            (item.name && item.name.search(this.props.pattern) >= 0)
        );
    }

    render() {
        let items = this.props.items
            .filter((item, key) => this.filterItem(item));
        if (!items.length) {
            return null
        }

        return [
            <WeaponsHeader
                    key="header"
                    name={this.props.name}/>,
            <tbody key="body">
            {items.map((item, key) => {
                return <WeaponsRow key={key} {...item}/>
            })}
            </tbody>
        ];
    }
};

class WeaponsTable extends Reflux.Component
{
    constructor(props) {
        super(props);
        this.store = DataStore;
        this.storeKeys = ['weapons', 'search'];
    }

    componentDidMount() {
        if (!this.state.weapons.length) {
            listDataActions.fetchItems('weapons');
        }
    }

    render() {
        let pattern = new RegExp(this.state.search, "i");
        return <div>
            <h2 className="icon fa-cutlery">Weapons</h2>
            <table className="nice-table condensed bordered responsive">
                {this.state.weapons
                    .map((set, key) => {
                        return <WeaponsBody
                            key={key}
                            pattern={pattern}
                            {...set}/>
                    })
                }
            </table>
        </div>
    }
}

export default WeaponsTable;