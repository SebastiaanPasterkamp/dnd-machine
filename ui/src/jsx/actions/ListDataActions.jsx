import React from 'react';
import Reflux from 'reflux';
import _ from 'lodash';

import ReportingActions from './ReportingActions.jsx';

function jsonOrBust(response) {
    if (response.ok) {
        return response.json();
    }
    throw response.statusText;
};

var ListDataActions = Reflux.createActions({
    "setState": {},
    "doLogin": {children: ['completed', 'failed']},
    "doLogout": {children: ['completed', 'failed']},
    "fetchItems": {children: ['completed', 'failed']},
});
let throttledGet = {};

ListDataActions.fetchItems.listen((type, category=null) => {
    let path = '/' + _.filter([
        category, type, category ? 'api' : null]).join('/');

    if (!(path in throttledGet)) {
        throttledGet[path] = _.throttle((path, type) => {

            fetch(path, {
                credentials: 'same-origin',
                'headers': {
                    'X-Requested-With': 'XMLHttpRequest'
                }
            })
            .then(jsonOrBust)
            .then((response) => {
                ListDataActions.fetchItems.completed({
                    [type]: response
                });
            })
            .catch((error) => {
                console.log(error);
                ListDataActions.fetchItems.failed(type, error);
            });
        }, 1000, {leading: true, trailing: false});
    }

    throttledGet[path](path, type);
});

ListDataActions.doLogin.listen((credentials, success, failure) => {
    fetch('/login', {
        credentials: 'same-origin',
        method: 'POST',
        'headers': {
            'X-Requested-With': 'XMLHttpRequest',
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(credentials)
    })
    .then(jsonOrBust)
    .then((response) => {
        ListDataActions.doLogin.completed(response);
        ListDataActions.fetchItems.completed({
            current_user: response
        });

        ReportingActions.showMessage(
            'good',
            'Login successful',
            'Login',
            5
        );

        if (success) {
            success(content);
        }
    })
    .catch((error) => {
        console.log(error);

        ReportingActions.showMessage(
            'bad',
            'Login failed',
            'Login',
            10
        );

        ListDataActions.doLogin.failed(error);

        if (failure) {
            failure();
        }
    });
});

ListDataActions.doLogout.listen((callback) => {
    fetch('/logout', {
        credentials: 'same-origin',
        'headers': {
            'X-Requested-With': 'XMLHttpRequest'
        }
    })
    .then(jsonOrBust)
    .then((response) => {
        ListDataActions.doLogout.completed(response);
        ListDataActions.fetchItems.completed({
            current_user: null
        });

        ReportingActions.showMessage(
            'good',
            'Logout successful',
            'Logout',
            5
        );

        if (callback) {
            callback();
        }
    })
    .catch((error) => {
        console.log(error);
        ListDataActions.doLogout.failed(error);
        ListDataActions.fetchItems.completed({
            current_user: null
        });

        ReportingActions.showMessage(
            'bad',
            'Logout failed',
            'Logout',
            10
        );

        if (callback) {
            callback();
        }
    });
});

export default ListDataActions;
