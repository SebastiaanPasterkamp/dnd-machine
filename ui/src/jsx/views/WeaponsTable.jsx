import React from 'react';
import {Link} from 'react-router-dom';

import ItemStore from '../mixins/ItemStore.jsx';

import LazyComponent from '../components/LazyComponent.jsx';
import ButtonField from '../components/ButtonField.jsx';
import Damage from '../components/Damage.jsx';
import Reach from '../components/Reach.jsx';
import Coinage from '../components/Coinage.jsx';

class WeaponsHeader extends LazyComponent
{
    render() {
        return <thead>
            <tr>
                <th>{this.props.name}</th>
                <th>Damage</th>
                <th>Range</th>
                <th>Cost</th>
                <th>Properties</th>
                <th>Actions</th>
            </tr>
        </thead>
    }
};

class WeaponsRow extends LazyComponent
{
    constructor(props) {
        super(props);
        this.properties = [
            {'code': 'ammunition', 'label': 'Ammunition'},
            {'code': 'finesse', 'label': 'Finesse'},
            {'code': 'heavy', 'label': 'Heavy'},
            {'code': 'light', 'label': 'Light'},
            {'code': 'loading', 'label': 'Loading'},
            {'code': 'reach', 'label': 'Reach'},
            {'code': 'special', 'label': 'Special'},
            {'code': 'thrown', 'label': 'Thrown'},
            {'code': 'two-handed', 'label': 'Two-Handed'},
            {'code': 'versatile', 'label': 'Versatile'},
        ];
    }

    render() {
        return <tr
                data-name={this.props.name}>
            <td>{this.props.name}</td>
            <td><Damage {...this.props.damage}/></td>
            <td>{this.props.range
                ? <Reach {...this.props.range}/>
                : null
            }</td>
            <td>{this.props.cost
                ? <Coinage {...this.props.cost} extended="1" />
                : null
            }</td>
            <td>{this.props.property
                ? <ul>{this.properties.map((prop) => {
                    return _.indexOf(this.props.property, prop.code) >= 0
                        ? <li key={prop.code}>{prop.label}</li>
                        : null
                })}</ul>
                : null
            }</td>
            <td>{this.props.id != null ?
                <Link
                    to={"/items/weapons/edit/" + this.props.id}
                    className="nice-btn-alt icon fa-pencil">
                    Edit
                </Link> : null
            }</td>
        </tr>
    }
};

class WeaponsBody extends LazyComponent
{
    filterItem(item) {
        return (
            (item.name && item.name.search(this.props.pattern) >= 0)
        );
    }

    render() {
        let items = this.props.items
            .filter((item, key) => this.filterItem(item));
        if (!items.length) {
            return null
        }

        return [
            <WeaponsHeader
                    key="header"
                    name={this.props.name}/>,
            <tbody key="body">
            {items.map((item, key) => {
                return <WeaponsRow key={key} {...item}/>
            })}
            </tbody>
        ];
    }
};

class WeaponsTable extends LazyComponent
{
    render() {
        let pattern = new RegExp(this.props.search, "i");

        return <div>
            <h2 className="icon fa-cutlery">Weapons</h2>
            <table className="nice-table condensed bordered responsive">
                {this.props.weapons
                    .map((set, key) => {
                        return <WeaponsBody
                            key={key}
                            pattern={pattern}
                            {...set}/>
                    })
                }
            </table>
        </div>
    }
}

export default ItemStore(WeaponsTable, ['weapons', 'search']);
