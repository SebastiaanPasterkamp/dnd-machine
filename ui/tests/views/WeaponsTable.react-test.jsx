import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MockRouter from 'react-mock-router';

import ListDataActions from 'actions/ListDataActions';
import {
    mockedApi,
    damage_types,
    weapon_properties,
    weapon_types,
} from '../__mocks__';

import WeaponsTable from 'views/WeaponsTable.jsx';

describe('WeaponsTable', () => {
    const fullProps = {
        weapons: {
            1: {
                id: 1,
                name: "Dagger",
                damage: {
                    dice_count: 1,
                    dice_size: 4,
                    type: "piercing",
                },
                cost: { gp: 2 },
                property: [ "thrown", "finesse", "light" ],
                range: { max: 60, min: 20 },
                type: "simple melee",
                weight: { lb: 1.0 }
            },
        },
    };

    beforeAll(() => {
        fetch.mockImplementation( mockedApi({
            damage_types,
            weapon_properties,
            weapon_types,
            user: {},
            'weapon/api': fullProps.weapons,
        }) );
        ListDataActions.fetchItems('damage_types', 'items');
        ListDataActions.fetchItems('weapon_properties', 'items');
        ListDataActions.fetchItems('weapon_types', 'items');
    })

    afterAll(() => {
        fetch.resetMocks()
    })

    describe('render', () => {
        it('should work without props', () => {
            const tree = renderer.create(
                <MockRouter location={{pathname: `/items/weapon/list`}}>
                    <WeaponsTable />
                </MockRouter>
            );

            expect(tree).toMatchSnapshot();
        });

        it('should work with all props', () => {
            const tree = renderer.create(
                <MockRouter location={{pathname: `/items/weapon/list`}}>
                    <WeaponsTable
                        {...fullProps}
                    />
                </MockRouter>
            );

            expect(tree).toMatchSnapshot();
        });
    });

    describe('when searching', () => {
        let wrapper;

        beforeAll(() => {
            wrapper = mount(
                <MockRouter location={{pathname: `/items/weapon/list`}}>
                    <WeaponsTable
                        {...fullProps}
                    />
                </MockRouter>
            );
        });

        it('should match', () => {
            expect(wrapper
                .find('.weapons-table .base-link-group')
                .length
            ).toBe(1);

            wrapper
                .find('input[placeholder="Name..."]')
                .simulate('change', {target: {value: 'dagger'}});
            wrapper
                .find('li[data-value="thrown"] input')
                .simulate('change', {target: {checked: true}});
            wrapper
                .find('li[data-value="simple melee"] input')
                .simulate('change', {target: {checked: true}});
            wrapper
                .find('input[placeholder="Dice count"]')
                .simulate('change', {target: {value: '1'}});

            expect(wrapper
                .find('.weapons-table .base-link-group')
                .length
            ).toBe(1);
        });

        it('should filter', () => {
            expect(wrapper
                .find('.weapons-table .base-link-group')
                .length
            ).toBe(1);

            wrapper
                .find('input[placeholder="Name..."]')
                .simulate('change', {target: {value: 'mace'}});

            expect(wrapper
                .find('.weapons-table .base-link-group')
                .length
            ).toBe(0);
        });
    });
});
