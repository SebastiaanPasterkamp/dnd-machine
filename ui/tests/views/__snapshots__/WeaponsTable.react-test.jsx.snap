// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeaponsTable render should work with all props 1`] = `
<div
  className="weapons-table"
>
  <h2
    className="icon fa-cutlery"
  >
    Weapons
  </h2>
  <div
    className="weapons-table--search"
  >
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Name
      </span>
      <input
        className="nice-form-control"
        disabled={undefined}
        id="name"
        onChange={[Function]}
        onKeyPress={null}
        placeholder="Name..."
        type="text"
        value=""
      />
    </div>
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Properties
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          All Properties
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value="ammunition"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Ammunition
            </label>
          </li>
          <li
            className={null}
            data-value="finesse"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Finesse
            </label>
          </li>
          <li
            className={null}
            data-value="heavy"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Heavy
            </label>
          </li>
          <li
            className={null}
            data-value="light"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Light
            </label>
          </li>
          <li
            className={null}
            data-value="loading"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Loading
            </label>
          </li>
          <li
            className={null}
            data-value="reach"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Reach
            </label>
          </li>
          <li
            className={null}
            data-value="ranged"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Ranged
            </label>
          </li>
          <li
            className={null}
            data-value="special"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Special
            </label>
          </li>
          <li
            className={null}
            data-value="thrown"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Thrown
            </label>
          </li>
          <li
            className={null}
            data-value="two-handed"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Two-Handed
            </label>
          </li>
          <li
            className={null}
            data-value="versatile"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Versatile
            </label>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Weapon Types
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          All Weapon Types
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value="simple melee"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Simple Melee Weapon
            </label>
          </li>
          <li
            className={null}
            data-value="simple ranged"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Simple Ranged Weapon
            </label>
          </li>
          <li
            className={null}
            data-value="martial melee"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Martial Melee Weapon
            </label>
          </li>
          <li
            className={null}
            data-value="martial ranged"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Martial Ranged Weapon
            </label>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Damage
      </span>
      <input
        className="nice-form-control"
        disabled={undefined}
        onChange={[Function]}
        onKeyPress={null}
        placeholder="Dice count"
        type="number"
        value=""
      />
      <span
        className="nice-input-addon"
      >
        d
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          Dice Sizes
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value={4}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              4
            </label>
          </li>
          <li
            className={null}
            data-value={6}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              6
            </label>
          </li>
          <li
            className={null}
            data-value={8}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              8
            </label>
          </li>
          <li
            className={null}
            data-value={10}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              10
            </label>
          </li>
          <li
            className={null}
            data-value={12}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              12
            </label>
          </li>
        </ul>
      </div>
      <span
        className="nice-input-addon"
      >
        Type
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          All Damage Types
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value="acid"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Acid
            </label>
          </li>
          <li
            className={null}
            data-value="bludgeoning"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Bludgeoning
            </label>
          </li>
          <li
            className={null}
            data-value="cold"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Cold
            </label>
          </li>
          <li
            className={null}
            data-value="fire"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Fire
            </label>
          </li>
          <li
            className={null}
            data-value="force"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Force
            </label>
          </li>
          <li
            className={null}
            data-value="lightning"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Lightning
            </label>
          </li>
          <li
            className={null}
            data-value="necrotic"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Necrotic
            </label>
          </li>
          <li
            className={null}
            data-value="piercing"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Piercing
            </label>
          </li>
          <li
            className={null}
            data-value="poison"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Poison
            </label>
          </li>
          <li
            className={null}
            data-value="psychic"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Psychic
            </label>
          </li>
          <li
            className={null}
            data-value="radiant"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Radiant
            </label>
          </li>
          <li
            className={null}
            data-value="slashing"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Slashing
            </label>
          </li>
          <li
            className={null}
            data-value="thunder"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Thunder
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <table
    className="nice-table condensed bordered responsive"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Damage
        </th>
        <th>
          Range
        </th>
        <th>
          Cost
        </th>
        <th>
          Properties
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        data-id={1}
      >
        <th>
          Dagger
          <div
            className="base-link-group nice-btn-group"
          >
            <a
              className="cursor-pointer nice-btn-alt icon fa-eye"
              href={undefined}
              onClick={[Function]}
            >
              View
            </a>
          </div>
        </th>
        <td>
          <div
            className="dice-notation inline capitalize"
          >
            <code>
              1
              d
              4
              
            </code>
            piercing
          </div>
        </td>
        <td>
          <div
            className="reach inline"
          >
            20
             ft. / 
            60
             ft.
          </div>
        </td>
        <td>
          <div
            className="coinage capitalize"
          >
            2 Gold
          </div>
        </td>
        <td>
          <ul>
            <li>
              <div
                className="list-label inline"
              >
                Simple Melee Weapon
              </div>
            </li>
            <li>
              <div
                className="list-label inline"
              >
                Thrown
              </div>
            </li>
            <li>
              <div
                className="list-label inline"
              >
                Finesse
              </div>
            </li>
            <li>
              <div
                className="list-label inline"
              >
                Light
              </div>
            </li>
          </ul>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <td
          colSpan={5}
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WeaponsTable render should work without props 1`] = `
<div
  className="weapons-table"
>
  <h2
    className="icon fa-cutlery"
  >
    Weapons
  </h2>
  <div
    className="weapons-table--search"
  >
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Name
      </span>
      <input
        className="nice-form-control"
        disabled={undefined}
        id="name"
        onChange={[Function]}
        onKeyPress={null}
        placeholder="Name..."
        type="text"
        value=""
      />
    </div>
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Properties
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          All Properties
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value="ammunition"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Ammunition
            </label>
          </li>
          <li
            className={null}
            data-value="finesse"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Finesse
            </label>
          </li>
          <li
            className={null}
            data-value="heavy"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Heavy
            </label>
          </li>
          <li
            className={null}
            data-value="light"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Light
            </label>
          </li>
          <li
            className={null}
            data-value="loading"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Loading
            </label>
          </li>
          <li
            className={null}
            data-value="reach"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Reach
            </label>
          </li>
          <li
            className={null}
            data-value="ranged"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Ranged
            </label>
          </li>
          <li
            className={null}
            data-value="special"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Special
            </label>
          </li>
          <li
            className={null}
            data-value="thrown"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Thrown
            </label>
          </li>
          <li
            className={null}
            data-value="two-handed"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Two-Handed
            </label>
          </li>
          <li
            className={null}
            data-value="versatile"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Versatile
            </label>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Weapon Types
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          All Weapon Types
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value="simple melee"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Simple Melee Weapon
            </label>
          </li>
          <li
            className={null}
            data-value="simple ranged"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Simple Ranged Weapon
            </label>
          </li>
          <li
            className={null}
            data-value="martial melee"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Martial Melee Weapon
            </label>
          </li>
          <li
            className={null}
            data-value="martial ranged"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Martial Ranged Weapon
            </label>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="nice-control-group"
    >
      <span
        className="nice-input-addon"
      >
        Damage
      </span>
      <input
        className="nice-form-control"
        disabled={undefined}
        onChange={[Function]}
        onKeyPress={null}
        placeholder="Dice count"
        type="number"
        value=""
      />
      <span
        className="nice-input-addon"
      >
        d
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          Dice Sizes
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value={4}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              4
            </label>
          </li>
          <li
            className={null}
            data-value={6}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              6
            </label>
          </li>
          <li
            className={null}
            data-value={8}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              8
            </label>
          </li>
          <li
            className={null}
            data-value={10}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              10
            </label>
          </li>
          <li
            className={null}
            data-value={12}
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              12
            </label>
          </li>
        </ul>
      </div>
      <span
        className="nice-input-addon"
      >
        Type
      </span>
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          onClick={[Function]}
        >
          All Damage Types
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={null}
        >
          <li
            className={null}
            data-value="acid"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Acid
            </label>
          </li>
          <li
            className={null}
            data-value="bludgeoning"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Bludgeoning
            </label>
          </li>
          <li
            className={null}
            data-value="cold"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Cold
            </label>
          </li>
          <li
            className={null}
            data-value="fire"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Fire
            </label>
          </li>
          <li
            className={null}
            data-value="force"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Force
            </label>
          </li>
          <li
            className={null}
            data-value="lightning"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Lightning
            </label>
          </li>
          <li
            className={null}
            data-value="necrotic"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Necrotic
            </label>
          </li>
          <li
            className={null}
            data-value="piercing"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Piercing
            </label>
          </li>
          <li
            className={null}
            data-value="poison"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Poison
            </label>
          </li>
          <li
            className={null}
            data-value="psychic"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Psychic
            </label>
          </li>
          <li
            className={null}
            data-value="radiant"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Radiant
            </label>
          </li>
          <li
            className={null}
            data-value="slashing"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Slashing
            </label>
          </li>
          <li
            className={null}
            data-value="thunder"
          >
            <label>
              <input
                checked={false}
                disabled={undefined}
                onChange={[Function]}
                type="checkbox"
              />
              Thunder
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <table
    className="nice-table condensed bordered responsive"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Damage
        </th>
        <th>
          Range
        </th>
        <th>
          Cost
        </th>
        <th>
          Properties
        </th>
      </tr>
    </thead>
    <tbody />
    <tbody>
      <tr>
        <td
          colSpan={5}
        />
      </tr>
    </tbody>
  </table>
</div>
`;
